jax/experimental/optix.py

```diff
@@ -404,34 +444,50 @@ def apply_updates(params, updates):
   return tree_multimap(lambda p, u: p + u, params, updates)
 
 
-### Aliases for popular optimizers. ###
+###
+# Aliases for popular optimizers.
 
 
-def sgd(learning_rate, momentum=0., nesterov=False):
+def sgd(learning_rate: float,
+        momentum: float = 0.,
+        nesterov: bool = False) -> InitUpdate:
   return chain(
       trace(decay=momentum, nesterov=nesterov),
-      scale(-learning_rate))
+      scale(-learning_rate),
+  )
 
 
-def noisy_sgd(learning_rate, eta=0.01, gamma=0.55, seed=42):
+def noisy_sgd(learning_rate: float,
+              eta: float = 0.01,
+              gamma: float = 0.55,
+              seed: int = 0) -> InitUpdate:
   return chain(
       trace(decay=0., nesterov=False),
       scale(-learning_rate),
-      add_noise(eta, gamma, seed))
+      add_noise(eta, gamma, seed),
+  )
 
 
-def adam(learning_rate, b1=0.9, b2=0.999, eps=1e-8):
+def adam(learning_rate: float,
```

Comment:
Should we have `AdamState` (etc) for the common cases (e.g. `AdamState = Sequence[ScaleByAdamState, ScaleState, ScaleState]`).

Reply:
Sure, I'll add that now.

Reply:
I'll leave that for a follow-up. I think we should improve the other types before we do that -- e.g. by making InitUpdate parametric/generic w.r.t. the state, so that we can have InitUpdate[Adam] rather than shoving unions of things into InitUpdate. WDYT?

Reply:
Sure, I think that will require quite a bit of refactoring though (e.g. I'm pretty sure I've tried to subclass NamedTuple and Generic before and it failed because of a metaclass conflict)? Up to you if you think its worth it.

Reply:
Ack -- will return to this at some point if I have time.